context/context.go

```diff
@@ -159,6 +159,23 @@ func (c *Context) Del(key string) error {
 	return c.db.Delete([]byte(key), &opt.WriteOptions{Sync: true})
 }
 
+// get everything in the db to dump it out for backup purposes
+func (c *Context) GetAll() (map [string]Route, error) {
+	golinks := map[string]Route{}
+	iter := c.db.NewIterator(nil, nil)
+	for iter.Next() {
+		key := iter.Key()
+		val := iter.Value()
+		rt := &Route{}
+		if err := rt.read(bytes.NewBuffer(val)); err != nil {
+			return nil, err
+		}
+		golinks[string(key[:])] = *rt
+	}
+
+	return golinks, nil
```

Comment:
I think leveldb iterators use that pattern where error on an iteration is stored in the iterator. So it is possible for `Next` to return false due to an error. It's typical to check `it.Error()` or to return that as the `err`.